src/audio/codec_adapter/codec_adapter.c

```diff
@@ -434,6 +434,9 @@ static int codec_adapter_copy(struct comp_dev *dev)
 		bytes_to_process -= codec->cpd.consumed;
 		processed += codec->cpd.consumed;
 		produced += codec->cpd.produced;
+
+		audio_stream_produce(&local_buff->stream, produced);
```

Comment:
I don't follow your reasoning here.

If you update in each iteration the `processed` and `produced` and then update the pointers in `local_buff` and `source `buffer, wouldn't that mean that you process some bytes twice?

In the first iteration you decode x1 bytes and get y1 bytes = >` processed = x1` and` produced = y1`.
In next iteration you decode x2 and get y2 => `processed = x1+x2`   and `produced  = y1 + y2`.
That means that x1 and y1 are taken into account twice when updating the buffers pointers.

Reply:
Good catch. Actually what I meant is this:

```
-               audio_stream_produce(&local_buff->stream, produced);
-               comp_update_buffer_consume(source, processed);
+               audio_stream_produce(&local_buff->stream, codec->cpd.produced);
+               comp_update_buffer_consume(source,codec->cpd.consumed);
```